solidity/dashboard/src/components/Timeline.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { useContext } from "react"
+
+const TimelineContext = React.createContext({
+  // May be useful in the future, eg. if we would like to be able to display the
+  // timeline events on alternating sides.
+  position: "left",
+})
+
+const useTimelineContext = () => {
+  const context = useContext(TimelineContext)
+  if (!context) {
+    throw new Error("TimelineContext used outside of Timeline component")
+  }
+  return context
+}
+
+const Timeline = ({ children, position = "left" }) => {
+  return (
+    <TimelineContext.Provider value={{ position: position }}>
+      <div className="timeline">{children}</div>
+    </TimelineContext.Provider>
+  )
+}
+const TimelineElement = ({ position, children }) => {
+  const { position: timelinePosition } = useTimelineContext()
+  const _position = position || timelinePosition
+  return (
+    <div className={`timeline__element timeline__element--${_position}`}>
+      {children}
+    </div>
+  )
+}
+Timeline.Element = TimelineElement
+
+Timeline.ElementBreakpoint = ({
+  children,
+  className = "",
+  // start, center
+  // TODO add support for a start position.
```

Comment:
Just pointing out that there is a `TODO` here. Not sure if it's done, or you planning to do it later, but if it's done then we should remove it.

Reply:
I'm going to address it in follow-up work. I'll add this when I work on a [timeline in modal](https://www.figma.com/file/Su61MwHLrQn3ld24q85pMx/Token-Dashboard-dApp?node-id=9934%3A33738).